joint_limits_interface/include/joint_limits_interface/joint_limits_rosparam.h

```diff
@@ -168,6 +168,78 @@ inline bool getJointLimits(const std::string& joint_name, const ros::NodeHandle&
   return true;
 }
 
+/**
+ * \brief Populate a SoftJointLimits instance from the ROS parameter server.
+ *
+ * It is assumed that the following parameter structure is followed on the provided NodeHandle. Unspecified parameters
+ * are simply not added to the joint limits specification.
+ * \code
+ * soft_joint_limits:
+ *   foo_joint:
+ *     soft_lower_limit: 0.0
+ *     soft_upper_limit: 1.0
+ *     k_position: 10.0
+ *     k_velocity: 10.0
+ * \endcode
+ *
+ * This specification is similar to the specification of the safety_controller tag in the URDF, adapted to the parameter server.
+ *
+ * \param[in] joint_name Name of joint whose limits are to be fetched.
+ * \param[in] nh NodeHandle where the joint limits are specified.
+ * \param[out] soft_limits Where soft joint limit data gets written into. Limits specified in the parameter server will overwrite
+ * existing values. Values in \p soft_limits not specified in the parameter server remain unchanged.
+ * \return True if a limits specification is found (ie. the \p soft_joint_limits/joint_name parameter exists in \p nh), false otherwise.
+ */
+inline bool getSoftJointLimits(const std::string& joint_name, const ros::NodeHandle& nh, SoftJointLimits& soft_limits)
+{
+  // Node handle scoped where the soft joint limits are defined
+  ros::NodeHandle soft_limits_nh;
+  try
+  {
+    const std::string soft_limits_namespace = "soft_joint_limits/" + joint_name;
+    if (!nh.hasParam(soft_limits_namespace))
+    {
+      ROS_DEBUG_STREAM("No soft joint limits specification found for joint '" << joint_name <<
+                       "' in the parameter server (namespace " << nh.getNamespace() + "/" + soft_limits_namespace << ").");
+      return false;
+    }
+    soft_limits_nh = ros::NodeHandle(nh, soft_limits_namespace);
+  }
+  catch(const ros::InvalidNameException& ex)
+  {
+    ROS_ERROR_STREAM(ex.what());
+    return false;
+  }
+
+  // Soft position limits
+  double soft_lower_limit;
```

Comment:
I'd prefer using `soft_limits_nh.hasParam(XYZ)` and calling `getParam` to copy directly onto the limits object. This saves us a line for each move and simplifies the code.

Reply:
Reasonable enough.

This makes me wonder whether `getParam` is guaranteed not to modify the value of the output variable if the parameter does not exist, which would further simplify the code. A quick test suggests that it doesn't, but since it's not mentioned in the documentation at all, I guess it's better not to rely on that behavior.